source/common/http/headers.h

```diff
@@ -64,6 +64,12 @@ class CustomHeaderValues {
   const LowerCaseString ContentEncoding{"content-encoding"};
   const LowerCaseString Etag{"etag"};
   const LowerCaseString GrpcAcceptEncoding{"grpc-accept-encoding"};
+  const LowerCaseString IfMatch{"if-match"};
+  const LowerCaseString IfNoneMatch{"if-none-match"};
+  const LowerCaseString IfModifiedSince{"if-modified-since"};
+  const LowerCaseString IfUnmodifiedSince{"if-unmodified-since"};
+  const LowerCaseString IfRange{"if-range"};
+  const LowerCaseString LastModified{"last-modified"};
```

Comment:
Shall we register them as static inline headers instead of using `get` through out the implementation?

Reply:
I could add a `common.h` file to the cache tree that registers all the used headers in this module to avoid registering them in every file that uses them, what do you think?

Reply:
But I guess this won't work without making the handles static which is against the style guide.

Reply:
There is also no `setReferenceKeyInline` to replace `setReferenceKey(Http::CustomHeaders::get().IfModifiedSince, date)`.

Reply:
OK, can you do it as a follow up?

Reply:
Can you elaborate further on what you are suggesting as a follow up?